src/components/backup/RestoreCloudStep.tsx

```diff
@@ -158,11 +164,15 @@ export default function RestoreCloudStep() {
       });
       if (status === RestoreCloudBackupResultStates.success) {
         // Store it in the keychain in case it was missing
-        const hasSavedPassword = await getLocalBackupPassword();
-        if (!hasSavedPassword) {
+        if (backupsStore.getState().storedPassword !== pwd) {
```

Comment:
this will prompt a faceID if the storedPassword is either a) not set, or b) different than the restoration password